src/goto-programs/json_goto_trace.cpp

```diff
@@ -141,14 +145,17 @@ void convert(
         json_output["thread"]=json_numbert(std::to_string(step.thread_nr));
         json_output["outputID"]=json_stringt(id2string(step.io_id));
 
+        // Recovering the mode from the function
+        symbolt function_name=ns.lookup(source_location.get_function());
```

Comment:
Could `source_location.get_function()` ever be nil, or otherwise not converted? If so this lookup call should use one of the conditional-lookup variants that doesn't throw if the key is not found.